trunk/modules/python/dataset/reader.py

```diff
@@ -78,53 +83,109 @@ def openSICD(image, log=None) :
     wbData, cmplxData = pysix.six_sicd.read(image, schemaPaths)
     return (wbData, cmplxData)
 
-def readSICD(image, log=None) :
+def readSICD(image, log=None, **kwargs) :
     '''This method should read and prepare the data for training or testing.'''
     wbData, cmplxData = openSICD(image, log)
-    return convertPhaseAmp(wbData, log)
+    raw = kwargs.get('raw', False)
+    if not raw:
+        wbData = convertPhaseAmp(wbData, log)
+    else:
+        wbData = wbData, 'sicd'
+    return wbData
 
-def readSIDD(image, log=None) :
+def readSIDD(image, log=None, **kwargs) :
     '''This method should read a prepare the data for training or testing.'''
     raise NotImplementedError('Implement the datasetUtils.readSIDD() method')
 
 def openSIO(image, log=None) :
     import coda.sio_lite
+    if type(image) == np.ndarray:
+        return image
     return coda.sio_lite.read(image)
 
-def readSIO(image, log=None) :
-    imData = openSIO(image, log)
+def transformSIOData(imData, log=None):
+    ''' Apply phase amp conversion to image data '''
     if imData.dtype == np.complex64 :
         return convertPhaseAmp(imData, log)
     else :
         # TODO: this assumes the imData is already band-interleaved
         return imData
 
-def readNITF(image, log=None) :
+def readSIO(image, log=None, **kwargs) :
+    siodata = openSIO(image, log)
+    raw = kwargs.get('raw', False)
+    if not raw:
+        siodata = transformSIOData(siodata, log)
+    else:
+        siodata = siodata, '.sio'
+    return siodata
+
+def readerToNumpyArray(reader):
```

Comment:
Does this information not come back when we use "coda.sio_lite.read()"? We really only need to shape and dtype of the np.ndarray to get started. If this is returned, we should remove this additional logic.